Source/WebCore/html/MediaElementSession.cpp

```diff
@@ -1273,6 +1277,9 @@ void MediaElementSession::updateMediaUsageIfChanged()
         m_element.hasEverNotifiedAboutPlaying(),
         isOutsideOfFullscreen,
         isLargeEnoughForMainContent(MediaSessionMainContentPurpose::MediaControls),
+#if PLATFORM(COCOA) && !HAVE(CGS_FIX_FOR_RADAR_97530095)
+        m_element.isVisibleInViewport()
```

Comment:
Does this function get called whenever we scroll? i.e. whenever the element may have entered/exited the viewport?

Reply:
No it doesn't. The RenderView "is in viewport" tracking machinery, for which we already register for all video elements, only invokes its callback when it switches between in viewport / not in viewport.